packages/assets-controllers/src/TokensController.ts

```diff
@@ -667,144 +614,48 @@ export class TokensController extends BaseController<
     asset: Token,
     type: string,
     interactingAddress?: string,
-  ): Promise<AssetSuggestionResult> {
+  ): Promise<void> {
```

Comment:
If this wasn't already a breaking change I'd have avoided doing this, but the changes are already breaking.

The return value is not used at all by neither extension nor mobile.

Reply:
This makes more sense when looking at the before/after of how it's handled in the clients
- PR for extension https://github.com/MetaMask/metamask-extension/pull/18829/files#diff-f3e25ebd138f61f51874e39dcbfcc24b8aff46248bf5e7deebfc9561a88d3824R39
- Mobile is similar, no result is expected

Reply:
For reference, the counter argument is that it could be good practice to provide useful return data in all our API methods for the sake of future-proofing. Though given the level of complexity we've reached, I'd agree to simplify where we can as we can add this back when needed.

Naturally this wouldn't be possible if the return value ever reached the dApp, but currently the only result from the middleware is a boolean.

Reply:
The thing is that I'm not entirely sure what would be useful to return in this case. The `suggestedAssetMeta` or the `Token` from `addToken`.

If we do the second, and then we end up adding more types, it's going to turn into a bigger breaking change.

I think leaving it to return void is safest, as we probably won't need to update anything in every client if we eventually return something (even though it still counts as breaking change).